### PR TITLE
Pax keyword api access

### PR DESCRIPTION
**Pull request by nichenke**
This set of commits allows read and write API access to the PAX keyword space within tar.

### CODE REVIEW

File: libarchive/archive_write_set_format_pax.c [hunk 1]
```diff
@@ -66,6 +66,7 @@ struct pax {
 	unsigned flags;
 #define WRITE_SCHILY_XATTR       (1 << 0)
 #define WRITE_LIBARCHIVE_XATTR   (1 << 1)
+#define WRITE_GENERIC_KW	 (1 << 2)
```
Comment (heemstra) [inline libarchive/archive_write_set_format_pax.c]: Indentation is off if you care.  (need another space between W and the left paren)

File: libarchive/test/test_read_pax_header_kw.c [hunk 2]
```diff
@@ -28,8 +28,12 @@
  */
 
 #include "test.h"
+// /app/bin/libarchive_test -vv -r /app/libarchive/test test_write_pax_kw
```
Comment (heemstra) [inline libarchive/test/test_read_pax_header_kw.c]: does upstream care about the use of C++ style comments?

**Review by heemstra — APPROVED** [review]

**Comment by nichenke** [comment]
Thanks for the review -- I'll fix that stuff up or comment soon.

File: libarchive/archive_write_set_format_pax.c (hunk 1 above)
Comment (nichenke) [reply libarchive/archive_write_set_format_pax.c]: Will fix, thanks.

File: libarchive/test/test_read_pax_header_kw.c (hunk 2 above)
Comment (nichenke) [reply libarchive/test/test_read_pax_header_kw.c]: There are other files with C++ comments, so going to run with it... if & when upstream, we can fix too. I'm guessing more unit tests could be added too, only covering basics right now.

**Pull request merged**